pipdeptree.py

```diff
@@ -376,7 +379,13 @@ def filter(self, include, exclude):
                                 continue
                 else:
                     break
-
+        assert not exclude & set([n.key for n in m])
```

Comment:
Didn't understand why this assertion is required here.

Reply:
It is not really required ... I added a check while making changes to "fail fast".
In fact, all assertions in the "production" code should be checks for error conditions and return values to be honest ...